packages/next/src/server-build.ts

```diff
@@ -291,6 +293,30 @@ export async function serverBuild({
           protocol = 'https://';
         }
 
+        let origin: string | null = null;
+        if (protocol) {
+          const urlWithoutProtocol = rewrite.dest.substring(protocol.length);
+
+          // Find the first occurrence of any URL delimiter
+          const delimiters = ['/', '?', '#'];
+          let endIndex = -1;
+
+          for (const delimiter of delimiters) {
+            const index = urlWithoutProtocol.indexOf(delimiter);
+            if (index !== -1) {
+              endIndex = endIndex === -1 ? index : Math.min(endIndex, index);
+            }
+          }
```

Comment:
```suggestion
          for (const delimiter of delimiters) {
            const index = urlWithoutProtocol.indexOf(delimiter);
            if (index !== -1) {
              endIndex = index
              break
            }
          }
```
The delimiter ordering matters so seems like it should break on first occurrence instead of using min index and checking all un-necessarily.

Reply:
I thought the same with the code that I updated too, but the following url would break with these changes:

```
https://google.com/?param=thing/other-thing
```